pages/react-canvas-components.mdx

```diff
@@ -0,0 +1,256 @@
+
+```jsx animated-hexagons-v9j2q src/Canvas.js
+const [frameCount, setFrameCount] = React.useState(0);
+// this effect will increment frameCount by one at the next frame
+// and is called every time frameCount changes, creating a recursive loop
+// this makes the Canvas component re-render at every frame
+React.useEffect(() => {
+  const frameId = requestAnimationFrame(() => {
+    setFrameCount(frameCount + 1);
+  });
+  return () => {
+    cancelAnimationFrame(frameId);
+  };
+}, [frameCount, setFrameCount]);
+
+// here's the clearing at every render — at every frame.
+if (context !== null) {
+  context.clearRect(0, 0, actualWidth, actualHeight);
+}
+```
+
+But... the canvas is now white! It's because the hexagons are only rendered once – when `<RandomHexagons />` is first rendered. Yet, the canvas is cleared at every frame, erasing the hexagons on the second frame. Children components should also draw in the canvas at every frame. To force them to re-render, one solution is to share the `frameCount` from the canvas with the `Hexagon` component. It's achieved via a React context, like we did with `SharingContext`:
+
+```jsx
+const FrameContext = React.createContext(0);
+
+const Canvas = props => {
+  // [...]
+  return (
+    <SharingContext.Provider value={renderingContext}>
+      <FrameContext.Provider value={frameCount}>
+        <canvas />
+        {props.children}
+      </FrameContext.Provider>
+    </SharingContext.Provider>
+  );
+};
+```
+
+```jsx
+const Hexagon = props => {
+  const renderingContext = React.useContext(SharingContext);
+  const frameCount = React.useContext(FrameContext);
+  // drawing logic
+};
+```
+
+And now our hexagons are back on the screen! 🎉 While this works, the `FrameContext` has to be added in every child component. Two options are available to make sure those two contexts are being consumed together:
+
+- We can regroup them in a single context, sharing both the `renderingContext` and the `frameCount`. But the `frameCount` variable is not used in the child components, so it does not make sense to share its value.
+
+- Or, we can create a `useCanvas` hook to hide this complexity away! Even when consuming both React contexts, the hook can only return the canvas rendering context to the child components:
+
+```jsx
+export const useCanvas = () => {
+  React.useContext(FrameContext);
+  const renderingContext = React.useContext(CanvasContext);
+  return renderingContext;
+};
+```
+
+The `Hexagon` component logic now needs a small update to use this new hook:
+
+```jsx
+const Hexagon = props => {
+  const renderingContext = useCanvas();
+  // drawing logic
+};
+```
+
+### 🔄 Making the hexagons move
+
+If we want the hexagons to rotate, each hexagon should have its rotation angle change at every frame-by incrementing it by 1 for example-. To do this, the hexagons needs to _remember_ the rotation from the previous render. We will use a [React ref](/react-refs-evolution) to achieve this:
```

Comment:
- "should change its rotation angle at every frame"
- Add spaces around the dashes or, better, replace with parenthesis 
- Comma before "for example"